polar-core/src/partial/simplify.rs

```diff
@@ -28,8 +27,20 @@ pub fn simplify_bindings(bindings: Bindings) -> Bindings {
 
 pub struct Simplifier;
 impl Folder for Simplifier {
-    fn fold_term(&mut self, t: Term) -> Term {
-        fn maybe_unwrap_operation(o: &Operation) -> Option<Term> {
+    /// Deduplicate constraints.
+    fn fold_constraints(&mut self, c: Constraints) -> Constraints {
```

Comment:
This diff looks like a bigger change than it is.  I just moved maybe_unwrap_operation into `fold_operation` and removed the case for eliding a single item partial.